accounts-db/src/accounts.rs

```diff
@@ -1285,15 +1298,19 @@ impl Accounts {
         txs: &'a [SanitizedTransaction],
         execution_results: &'a [TransactionExecutionResult],
         load_results: &'a mut [TransactionLoadResult],
-        _rent_collector: &RentCollector,
         durable_nonce: &DurableNonce,
         lamports_per_signature: u64,
-    ) -> (
-        Vec<(&'a Pubkey, &'a AccountSharedData)>,
-        Vec<Option<&'a SanitizedTransaction>>,
-    ) {
+        preexecution_account_map: Option<&'a HashMap<Pubkey, AccountSharedData>>,
+    ) -> CollectedAccountsData<'a> {
         let mut accounts = Vec::with_capacity(load_results.len());
         let mut transactions = Vec::with_capacity(load_results.len());
+        let mut preexecution_account_datas = Vec::with_capacity(load_results.len());
```

Comment:
Yet more allocations that even validators (who don't need this). Are forced to perform. This is more bloat that will slow down performance-critical sections

Reply:
This allocation is vector of reference, so allocation of batch size which is 64 and reference is like 16 bytes should be really minimal and should not impact the performance that much. I can just remove the capacity and update the capacity only if the feature is enabled.

Reply:
It's an allocation that we won't need to do 99.99% of the time. The last thing the labs client needs is more pressure on the allocator

Reply:
Yes this will be fixed.